runtime/executor/tensor_parser.h

```diff
@@ -21,18 +21,34 @@ namespace executorch {
 namespace runtime {
 namespace deserialization {
 
+/// Data structure to hold key and data buffer for external data used
+/// in a method.
+struct NamedData {
+  const char* key;
+  FreeableBuffer buffer;
+};
+
+NamedData* get_data_by_key(const char* key, Span<NamedData> external_constants);
```

Comment:
This function searches a generic array of NamedData, and could be used in situations other than looking for external constants: a generic param name like `entries` would make that more clear.